ibis/formats/pyarrow.py

```diff
@@ -243,15 +244,30 @@ def convert_scalar(cls, scalar: pa.Scalar, dtype: dt.DataType) -> pa.Scalar:
             return scalar
 
     @classmethod
-    def convert_column(cls, column: pa.Array, dtype: dt.DataType) -> pa.Array:
+    def convert_column(
+        cls, column: pa.Array | pd.Series, dtype: dt.DataType
+    ) -> pa.Array:
+        if isinstance(column, pd.Series):
+            if dtype.is_uuid():
+                # pyarrow doesn't support UUIDs, so we need to convert them to strings
+                column = column.astype(str)
+            column = pa.Array.from_pandas(column)
         desired_type = PyArrowType.from_ibis(dtype)
         if column.type != desired_type:
             return column.cast(desired_type)
         else:
             return column
 
     @classmethod
-    def convert_table(cls, table: pa.Table, schema: Schema) -> pa.Table:
+    def convert_table(cls, table: pa.Table | pd.DataFrame, schema: Schema) -> pa.Table:
+        if isinstance(table, pd.DataFrame):
```

Comment:
We definitely do not want to mix pandas and pyarrow here, that's the primary reason these are implemented as separate classes.

Can you think of another way to solve this problem that doesn't require conflating the two conversion paths?

Reply:
Hmmm makes sense, will take a look.

Reply:
@cpcloud responded in a top-level comment.